alerts/system_alerts.libsonnet

```diff
@@ -59,6 +59,151 @@
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Kubelet could not be scraped or all kubelets disappeared from Prometheus target discovery.',
+            },
+          },
+          {
+            alert: 'KubeletTooManyPods',
+            expr: |||
+              kubelet_running_pod_count{%(kubeletSelector)s} > 100
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'Kubelet {{$labels.instance}} is running {{$value}} pods, close to the limit of 110.',
+            },
+          },
+          {
+            alert: 'KubeSchedulerDown',
+            expr: |||
+              absent(up{%(kubeSchedulerSelector)s} == 1)
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'There is no running Kubernetes scheduler. New pods are not being assigned to nodes.',
+            },
+          },
+          {
+            alert: 'KubeControllerManagerDown',
+            expr: |||
+              absent(up{%(kubeControllerManagerSelector)s} == 1)
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'There is no running Kubernetes controller-manager. Deployments, ReplicationControllers, etc. are not making progress.',
+            },
+          },
+          {
+            alert: 'KubeAPILatencyHigh',
+            expr: |||
+              apiserver_latency_seconds:quantile{%(kubeApiserverSelector)s,quantile="0.99",subresource!="log",verb!~"^(?:WATCH|WATCHLIST|PROXY|CONNECT)$"} > 1
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'The API server has a 99th percentile latency of {{ $value }} seconds for {{$labels.verb}} {{$labels.resource}}.',
+            },
+          },
+          {
+            alert: 'KubeAPILatencyHigh',
+            expr: |||
+              apiserver_latency_seconds:quantile{%(kubeApiserverSelector)s,quantile="0.99",subresource!="log",verb!~"^(?:WATCH|WATCHLIST|PROXY|CONNECT)$"} > 4
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'The API server has a 99th percentile latency of {{ $value }} seconds for {{$labels.verb}} {{$labels.resource}}.',
+            },
+          },
+          {
+            alert: 'KubeAPIErrorsHigh',
+            expr: |||
+              rate(apiserver_request_count{%(kubeApiserverSelector)s,code=~"^(?:5..)$"}[5m])
```

Comment:
I don't think this rule is actually correct, I copied it from our alerts, but from trying this doesn't seem right.

Reply:
I went with:

```
sum by (namespace, job) (rate(apiserver_request_count{code=~"5..",job="default/kubernetes"}[5m]))
 / 
sum  by (namespace, job) (rate(apiserver_request_count{job="default/kubernetes"}[5m])) * 100 
```

The sum helps tidy up the mess of dimensions on this metric.